File: AppGenerateFiles/security/Model/Security_Users.cs
using CAPA_DATOS;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
namespace DataBaseModel {
   public class Security_Users : EntityClass {
       [PrimaryKey(Identity = true)]
       public int? Id_User { get; set; }
       public string? Nombres { get; set; }
       public string? Estado { get; set; }
       public string? Descripcion { get; set; }
       public string? Password { get; set; }
       public string? Mail { get; set; }
       public string? Token { get; set; }
       public DateTime? Token_Date { get; set; }
       public DateTime? Token_Expiration_Date { get; set; }
       public int? Id_Sucursal { get; set; }
       public int? id_agente { get; set; }
       [ManyToOne(TableName = "Catalogo_Agentes", KeyColumn = "id_agente", ForeignKeyColumn = "id_agente")]
       public Catalogo_Agentes? Catalogo_Agentes { get; set; }
       [ManyToOne(TableName = "Catalogo_Sucursales", KeyColumn = "Id_Sucursal", ForeignKeyColumn = "Id_Sucursal")]
       public Catalogo_Sucursales? Catalogo_Sucursales { get; set; }
       [OneToMany(TableName = "Transaction_Contratos_Inversionistas", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Transaction_Contratos_Inversionistas>? Transaction_Contratos_Inversionistas { get; set; }
       [OneToMany(TableName = "Transaction_Facturas", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Transaction_Facturas>? Transaction_Facturas { get; set; }
       [OneToMany(TableName = "Security_Users_Roles", KeyColumn = "Id_User", ForeignKeyColumn = "Id_User")]
       public List<Security_Users_Roles>? Security_Users_Roles { get; set; }
   }
}

File: AppGenerateFiles/security/FrontModel/Security_Users.js
//@ts-check
import { EntityClass } from "../../WDevCore/WModules/EntityClass.js";
import { WAjaxTools, BasicStates } from "../../WDevCore/WModules/WComponentsTools.js";
import { ModelProperty } from "../../WDevCore/WModules/CommonModel.js";
import { Catalogo_Agentes }  from './Catalogo_Agentes.js'
import { Catalogo_Sucursales }  from './Catalogo_Sucursales.js'
import { Transaction_Contratos_Inversionistas }  from './Transaction_Contratos_Inversionistas.js'
import { Transaction_Facturas }  from './Transaction_Facturas.js'
import { Security_Users_Roles }  from './Security_Users_Roles.js'
class Security_Users extends EntityClass {
   constructor(props) {
       super(props, 'EntitySecurity');
       for (const prop in props) {
           this[prop] = props[prop];
       }
   }
   /**@type {Number}*/ Id_User;
   /**@type {String}*/ Nombres;
   /**@type {String}*/ Estado;
   /**@type {String}*/ Descripcion;
   /**@type {String}*/ Password;
   /**@type {String}*/ Mail;
   /**@type {String}*/ Token;
   /**@type {Date}*/ Token_Date;
   /**@type {Date}*/ Token_Expiration_Date;
   /**@type {Catalogo_Agentes} ManyToOne*/ Catalogo_Agentes;
   /**@type {Catalogo_Sucursales} ManyToOne*/ Catalogo_Sucursales;
   /**@type {Array<Transaction_Contratos_Inversionistas>} OneToMany*/ Transaction_Contratos_Inversionistas;
   /**@type {Array<Transaction_Facturas>} OneToMany*/ Transaction_Facturas;
   /**@type {Array<Security_Users_Roles>} OneToMany*/ Security_Users_Roles;
}
export { Security_Users }
